vmm/src/vm.rs

```diff
@@ -1146,6 +1147,20 @@ impl Vm {
             Error::ConfigureSystem(arch::Error::AArch64Setup(arch::aarch64::Error::SetupGic(e)))
         })?;
 
+        if vcpu_pmu {
```

Comment:
Can you add a public function in `CpuManager` to set the PMU for each VCPUs, and call that function here simply? That way you can avoid publishing `vcpus` of `CpuManager` and exposing the loop here.